File: static/js/model/wardrobe-paginator.js
var DUB = DUB || {};
DUB.model = DUB.model || {};

(function (ns) {
	ns.WardrobePaginator = function (data, pageSize) {
		var DEFAULT_PAGE_SIZE = 15;
	
		pageSize = typeof pageSize === 'undefined' ? DEFAULT_PAGE_SIZE : pageSize;
		
		var sets = [];
		var index = 0;

		for (var i = 0; i < data.length; i += pageSize) {
			sets.push(data.slice(i, i + Math.min(pageSize, data.length - i)));
		}
		
		return {
			next: function () {
				index = (index == sets.length - 1) ? index : ++index;
				return sets[index];
			},
			
			previous: function () {
				index = (index > 0) ? --index : index;
				return sets[index];
			},
			
			current: function () {
				return sets[index];
			}
		}
	}
}(DUB.model));

File: Assets/src/model/WardrobePaginator.cs
using System;
using System.Collections.Generic;

public class WardrobePaginator<T> {
	private const int DEFAULT_PAGE_SIZE = 15;

	private int index;
	private IList<T[]> sets;

	public WardrobePaginator(T[] data) {
		init(data, DEFAULT_PAGE_SIZE);
	}

	public WardrobePaginator(T[] data, int pageSize) {
		init(data, pageSize);
	}
	
	private void init(T[] data, int pageSize) {
		sets = new List<T[]>();
		index = 0;

		for (int i = 0; i < data.Length; i += pageSize) {
			int setSize = Math.Min(pageSize, data.Length - i);
			T[] currentSet = new T[setSize];
			Array.Copy(data, i, currentSet, 0, setSize);
			sets.Add(currentSet);
		}
	}


	public T[] Next() {
		if (sets.Count > 0) {
			index = (index == sets.Count - 1) ? index : index + 1;
			return sets[index];
		}
		return new T[] {};
	}

	public T[] Previous() {
		if (sets.Count > 0) {
			index = (index > 0) ? index - 1 : index;
			return sets[index];
		}
		return new T[] {};
	}

	public T[] Current() {
		if (sets.Count > 0) {
			return sets[index];
		}
		return new T[] {};
	}
}
